DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -296,23 +336,47 @@ function UninstallPackage
 
 function IsPackageInstalled
 {
+    [CmdletBinding(DefaultParameterSetName = 'RequiredVersion')]
+    [OutputType([Boolean])]
     param(
-        [Parameter(Position=0,Mandatory)][string]$pName,
-        [Parameter(Position=1)][string]$pVersion
+        [Parameter(Position=0, Mandatory)]
+        [string]$pName,
+        
+        [Parameter(ParameterSetName = 'RequiredVersion')]
+        [string]$pVersion,
+        
+        [Parameter(ParameterSetName = 'MinimumVersion')]
+        [string]$pMinimumVersion
     )
     Write-Verbose -Message "Start IsPackageInstalled $pName"
 
     $env:Path = [Environment]::GetEnvironmentVariable('Path','Machine')
-    Write-Verbose -Message "Path variables: $env:Path"
+    Write-Verbose -Message "Path variables: $($env:Path)"
 
     $installedPackages = Get-ChocoInstalledPackage
 
     if ($pVersion) {
-        Write-Verbose 'Comparing version'
-        $installedPackages = $installedPackages | Where-object { $_.Name -eq $pName -and $_.Version -eq $pVersion}
-    } else {
+        Write-Verbose 'Comparing required version'
+        $installedPackages = $installedPackages | Where-Object { $_.Name -eq $pName -and $_.Version -eq $pVersion}
+    }
+    elseif ($pMinimumVersion) {
+        Write-Verbose 'Comparing minimum version'
+        # version comparison can be done with [System.Version] but this lacks the ability to compare pre-release versions
+        # because of this limitation MinimumVersion cannot be used in conjuction with pre-release packages
+        $pre = ($pMinimumVersion -split "-")[1]
+        if ($pre) {
+            throw "MinimumVersion does not support comparing pre-releases, please use Version parameter instead"
+        }
+        $comparablePackages = $installedPackages | Where-Object { $_.Name -eq $pName} | ForEach-Object {
+            # ignoring pre-release as per above comment
+            $v = [System.Version](($_.Version -split "-")[0])
+            $_ | Add-Member -MemberType NoteProperty -Name ComparableVersion -Value $v -PassThru

```

Comment:
Are you absolutely sure this works? I have always been led to believe that you cannot modify an item on the pipeline.

Reply:
Yeah this normal use as far as I am aware, that is where pipelines are great at modifying objects and handing them to the next command. A quick example:

```powershell
$serviceObj = Get-Service audiosrv
Add-Member -InputObject $serviceObj -MemberType NoteProperty -Name MyDescription -Value "the sound thing"
$serviceObj | Select status,name,mydescription

Get-Service audiosrv | Add-Member -MemberType NoteProperty -Name MyDescription -Value "noisey" -PassThru | Select status,name,mydescription

```

both of these are equivalent but the later I would say is more "powershell-esque"